scipy/stats/_hypotests.py

```diff
@@ -130,3 +130,317 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+def _get_wilcoxon_distr(n):
+    """
+    Distribution of counts of the Wilcoxon ranksum statistic r_plus (sum of
+    ranks of positive differences). For fixed n, simulate all possible states
+    {0, 1}**n and compute the sum of the ranks over the indices that are equal
+    to one.
+    Code:
```

Comment:
In scipy.special, for precomputed values, we nowadays tend to write a routine that outputs the code in readily copypasteable form. Maybe here it would be possible to move the code from docstring to a function `def _precompute_wilcoxon_distr_table():` that outputs the table, so it's possible to verify the result easily.

Reply:
Also, it might be good to keep the big table of values out of the main code itself as it makes it rather hard to read. You can put this in e.g. `_wilcoxon_data.py`:

```python
COUNTS = {
    3: [1, 1, 1, 2, 1, 1, 1],
    4: [...],
    ...
}
```

and then in this file do

```python
from . import _wilcoxon_data

def _get_wilcoxon_distr(n):
    count = _wilcoxon_data.COUNTS.get(n)
    if count is None:
        raise ValueError(...)

    return ...
```

This also assists with @pv's suggestion as then you can just autogenerate the entire `_wilcoxon_data.py` file.

Reply:
@pv and @person142's suggestions have addressed, so I'm marking this as conversation as resolved.